### PR TITLE
Octanify

### PR DESCRIPTION
**Pull request by mrloop**
#### Description
[//]: # "Please include a summary of the change. Imagine you're trying to explain the changes to a reviewer."

Update component to Octance / Glimmer
Run codemods and than manually update to Glimmer. Maybe easier to review commit by commit.

Note docs need to be updated to angle bracket syntax, this will be done in another PR

### CODE REVIEW

File: addon/components/amount-input.js [hunk 1]
```diff
@@ -15,38 +15,41 @@ const KEY_CODE_COMMA = 188
...
+  get currency() {
+    return this.args.currency ?? 'EUR'
+  }
 
   /**
     A custom class applied on the input
     @argument inputClass
     @type String?
   */
-  inputClass: null,
```
Comment (mrloop) [inline addon/components/amount-input.js:34]: Keep comment for auto doc generation, run locally and see at http://localhost:4200/docs/api/components/amount-input

File: addon/components/amount-input.hbs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<div class="amount-input" ...attributes>
+  <span class="amount-input__currency {{ if @value "black" }}">{{this.currency}}</span>
```
Comment (herzzanu) [inline addon/components/amount-input.hbs:2]:
```suggestion
  <span class="amount-input__currency {{if @value "black"}}">{{this.currency}}</span>
```

File: addon/components/amount-input.hbs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+  <input
+    class="amount-input__value {{ @inputClass }} {{ if @disabled "disabled" }}"
```
Comment (herzzanu) [inline addon/components/amount-input.hbs:5]:
```suggestion
    class="amount-input__value {{@inputClass}} {{if @disabled "disabled"}}"
```

File: tests/dummy/app/controllers/application.js [hunk 4]
```diff
@@ -1,12 +1,12 @@
+import { action } from '@ember/object'
 import Controller from '@ember/controller'
 
-export default Controller.extend({
-  isExpanded: false,
-  value: null,
+export default class ApplicationController extends Controller {
+  isExpanded = false
+  value = null
```
Comment (herzzanu) [inline tests/dummy/app/controllers/application.js:6]:
```suggestion
  @tracked value = null
```

File: tests/dummy/app/controllers/application.js [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
 
-  actions: {
-    handleUpdate(value) {
-      this.set('value', value)
-    }
+  @action
+  handleUpdate(value) {
+    this.set('value', value)
```
Comment (herzzanu) [inline tests/dummy/app/controllers/application.js:10]:
```suggestion
    this.value = value;
```

**Review by herzzanu — COMMENTED** [review]
A couple of small comments, looks good otherwise 👍

File: addon/components/amount-input.hbs (hunk 2 above)
Comment (mrloop) [reply addon/components/amount-input.hbs]: Wonder why linting didn't pick this up, have updated. Will take a look at linting in another PR.

File: addon/components/amount-input.hbs (hunk 3 above)
Comment (mrloop) [reply addon/components/amount-input.hbs]: updated

File: tests/dummy/app/controllers/application.js (hunk 4 above)
Comment (mrloop) [reply tests/dummy/app/controllers/application.js]: This file is cruft, I've deleted it.

**Pull request merged**